osversion.go

```diff
@@ -2,16 +2,34 @@ package rig
 
 // OSVersion host operating system version information
 type OSVersion struct {
-	ID      string
-	IDLike  string
-	Name    string
-	Version string
+	ID          string
+	IDLike      string
+	Name        string
+	Version     string
+	extraFields map[string]string
 }
 
 // String returns a human readable representation of OSVersion
-func (o *OSVersion) String() string {
+func (o OSVersion) String() string {
 	if o.Name != "" {
 		return o.Name
 	}
 	return o.ID + " " + o.Version
 }
+
+// SetExtraField sets an extra field
+func (o *OSVersion) SetExtraField(key, value string) {
+	if o.extraFields == nil {
+		o.extraFields = make(map[string]string)
+	}
+	o.extraFields[key] = value
+}
+
+// GetExtraField retrieves an Extra Field value if it is set
+func (o OSVersion) GetExtraField(key string) (string, bool) {
+	if o.extraFields == nil {
+		return "", false
+	}
```

Comment:
Shouldn't need a nil-check: https://go.dev/play/p/we5mmPnepBH

Reply:
maps themselves can be nil, but you tend to run into the issue only when they are properties of structs such as in https://go.dev/play/p/_mnUcXHX4rs

Reply:
nil check now in place only before assignment in ParseOSReleaseFile()